homeassistant/components/matter/light.py

```diff
@@ -39,79 +71,328 @@ class MatterLight(MatterEntity, LightEntity):
 
     entity_description: MatterLightEntityDescription
 
+    def _supports_feature(
+        self, feature_map: int, feature: MatterColorControlFeatures
+    ) -> bool:
+        """Return if device supports given feature."""
+
+        return (feature_map & (1 << feature.value)) != 0
+
+    def _supports_color_mode(self, color_feature: MatterColorControlFeatures) -> bool:
+        """Return if device supports given color mode."""
+
+        if not self._supports_color():
+            return False
+
+        feature_map = self._device_type_instance.node.get_attribute(
+            self._device_type_instance.endpoint,
+            clusters.ColorControl,
+            clusters.ColorControl.Attributes.FeatureMap,
+        )
+
+        assert isinstance(feature_map.value, int)
+
+        return self._supports_feature(int(feature_map.value), color_feature)
+
+    def _supports_hs_color(self) -> bool:
+        """Return if device supports hs color."""
+
+        return self._supports_color_mode(MatterColorControlFeatures.HS)
+
+    def _supports_xy_color(self) -> bool:
+        """Return if device supports xy color."""
+
+        return self._supports_color_mode(MatterColorControlFeatures.XY)
+
+    def _supports_color_loop(self) -> bool:
```

Comment:
are there any effects known in Matter world as far as we are aware? Or is color loop the only one ?